### PR TITLE
feat: add support for proxy

### PR DESCRIPTION
**Pull request by xie-jinwei**
Octokit won't automatically read proxy setting from environment variables, causing this GitHub Action fail to run on a self-hosted GitHub Actions runner behind a proxy.

This PR fixes this. Now Octokit client will read proxy settings from environment variables on initialization (This PR is tested out on my own self-hosted GitHub Actions runner behind proxy)

Signed-off-by: xie-jinwei <jinwei.xie@intel.com>

### CODE REVIEW

**Comment by peter-evans** [comment]
Hi @xie-jinwei 

Thank you for contributing.

I actually have proxy support added to some of my other actions, but neglected to add it here. I would like the implementation to match other actions I maintain. Would you mind changing this PR to copy [this implementation](https://github.com/peter-evans/create-pull-request/blob/7380612b49221684fefa025244f2ef4008ae50ad/src/octokit-client.ts#L8-L23)? Or I can do it if you prefer?

**Comment by xie-jinwei** [comment]
> Hi @xie-jinwei
> 
> Thank you for contributing.
> 
> I actually have proxy support added to some of my other actions, but neglected to add it here. I would like the implementation to match other actions I maintain. Would you mind changing this PR to copy [this implementation](https://github.com/peter-evans/create-pull-request/blob/7380612b49221684fefa025244f2ef4008ae50ad/src/octokit-client.ts#L8-L23)? Or I can do it if you prefer?

Hi @peter-evans 
Thanks for your response, I have updated the implementation to yours version, but still support both http_proxy and https_proxy.

File: src/octokit-client.ts [hunk 1]
```diff
@@ -4,8 +4,28 @@ import {
   restEndpointMethods,
   RestEndpointMethodTypes
 } from '@octokit/plugin-rest-endpoint-methods'
+import {HttpProxyAgent} from 'http-proxy-agent'
+import {HttpsProxyAgent} from 'https-proxy-agent'
```
Comment (peter-evans) [inline src/octokit-client.ts:8]: I think you might need to add these to package.json
See https://github.com/peter-evans/create-pull-request/blob/7380612b49221684fefa025244f2ef4008ae50ad/package.json#L37

**Comment by peter-evans** [comment]
Thanks for switching the implementation.

Please run `npm run format` to fix the formatting error.

File: src/octokit-client.ts (hunk 1 above)
Comment (xie-jinwei) [reply src/octokit-client.ts:8]: Thanks, Peter, I have added it.

**Comment by xie-jinwei** [comment]
> Thanks for switching the implementation.
> 
> Please run `npm run format` to fix the formatting error.

Thanks, Peter, I have updated this.

**Comment by peter-evans** [comment]
Thank you again for contributing! Released as `v2.3.0` / `v2`.

**Pull request merged**